python/torch_mlir/csrc/base_lazy_backend/mlir_lowering_context.cpp

```diff
@@ -108,6 +109,10 @@ void TorchMlirLoweringContext::AddParameter(
 ComputationPtr TorchMlirLoweringContext::Build() {
   PRINT_FUNCTION();
 
+  // Since we mutated the types of some nodes to insert shape information, we
+  // must perform this pass to ensure tuples have up to date output types.
+  torch::jit::RefineTupleTypes(graph_);
```

Comment:
Seems this would require https://github.com/pytorch/pytorch/pull/76919 to be landed first?